Nos/Service/Relay/RelayService.swift

```diff
@@ -630,14 +630,21 @@ extension RelayService {
         }
     }
 
-    func relayAddresses(for user: CurrentUser) async -> [URL] {
-        await backgroundContext.perform { () -> [URL] in
+    func relayAddresses(for user: CurrentUser) async -> Set<URL> {
```

Comment:
Originally I changed the return type from an `Array` to `Set` because I was turning `userRelays` into a set in order to subtract the set of `Relay.searchOnly` relays. I then changed the logic in the function to look at the host of each relay URL instead to catch all subdomains of nostr.band. But I left the return value as a Set because we don't want duplicate relay addresses (although we also try to prevent writing duplicate relays to the db in the first place).